Add Auxiliary.GetMultiLinkedZone(tp), FaceupLinkMonsterFilter(c)

KoishiMint opened the pull request:
Added Auxiliary.GetMultiLinkedZone(tp) for multi-linked zone(zone linked by two or more link monsters).
Special thanks to Bilibili user "我才不会写代码呢(I won't write code any more)".

MonokuroInzanaito commented:
`local function f(c) return c:IsFaceup() and c:IsType(TYPE_LINK) end` What is this line for?
`mlz=slz & tlz | mlz` Which one goes first?

MonokuroInzanaito commented:
Shouldn't the 1st function be `Auxiliary.FaceupLinkMonsterFilter` or something like that?
`GetFaceupLinkMonster` is more likely a function returns a card group with all link monsters.

KoishiMint commented:
Updated function name and deleted the redundant line.
& goes first in order to calculate whether the monster being scanned increases multi-linked zone.
